docs/products/opensearch/howto/get-started-with-nodejs.rst

```diff
@@ -0,0 +1,528 @@
+
+::
+
+    avn user login <your.email@example.com>
+
+To verify that you have been authenticated run the following command
+
+::
+
+    avn user info
+
+The output will display your current user information. If you can see it, you're ready to create Aiven services. For more information on how to set up and use Aiven CLI check `its GitHub page <https://github.com/aiven/aiven-client>`_
+
+Now let’s create an OpenSearch cluster named *demo-open-search*, hosted in *Google Europe (Warsaw) Region*. A single-node setup will be sufficient for this tutorial, that's why we'll use *hobbyist* plan.
+
+::
+
+    avn service create demo-open-search     \
+        --service-type opensearch           \
+        --cloud google-europe-central2      \
+        --plan hobbyist;                    \
+    avn service wait demo-open-search
+
+Wait till the service reaches running state. At this point you’ll get a message in the terminal
+
+::
+
+    INFO	Service 'demo-open-search' state is now 'RUNNING'
+
+Next, we should find where the cluster is located in order to be able to connect. The parameter with the address is ``service_uri`` and can be requested through the command line.
+
+::
+
+    avn service get demo-open-search --format "The service is located at {service_uri}"
+
+You should see a URL as a command output.
+
+Great! Now we have the cluster set up and running. We’ll be using ``service_uri`` in just a couple of minutes.
+
+Setting up a NodeJS project
+----------------------------
+
+Make sure that you have both Node.js and npm installed. You can do it by checking the installed versions.
+
+::
+
+    node -v;    /
+    npm -v
+
+If you don’t have Node.js or npm installed, follow `these instructions <https://docs.npmjs.com/downloading-and-installing-node-js-and-npm>`_.
+
+To set up a new Node.js project run the following command in a place where you’d like to create a project and follow the instructions (you can accept default options).
+
+::
+
+    mkdir demo-open-search; /
+    cd demo-open-search;    /
+    npm init
+
+While we’re at it let’s also install `OpenSearch JavaScript client  <https://github.com/opensearch-project/opensearch-js>`_. At the moment of writing this tutorial the ``opensearch-js`` module is still not in the npm registry and we will install it directly from GitHub:
+
+::
+
+    npm install https://github.com/opensearch-project/opensearch-js
+
+Next add an empty ``index.js`` file into the project and open it with your favourite editor. We’ll be adding code into this file and running the methods from the command line.
+
+Start by adding the following lines of code to connect to the cluster. Use the ``service_uri`` you’ve got in the previous section.
+
+.. code:: javascript
+
+    const { Client } = require('@opensearch-project/opensearch')
+
+    const client = new Client({
+       node: "your-open-search-service-uri"
+    })
+
```

Comment:
Let's also explain what this code does: construct a client object that we can perform database operations on.